Dockerfile build can't run example missions

@clydemcqueen
I've recently decided to use this library to communicate with the BlueROV through ROS2. I built the docker file included in the repository and tried to execute the example mission provided in the documentation. It worked fine at first until all of a sudden there was a TCP error. I've copied the relevant error messages below. 

[controller_server-14] [INFO] [1675701766.523667448] [controller_server]: Passing new path to controller.
[ardusub-2] ERROR: Floating point exception - aborting
[ardusub-2] Running: sh dumpstack.sh 85909 >dumpstack.sh_ardusub.85909.out 2>&1
[ardusub-2] Failed
[ardusub-2] Running: sh dumpcore.sh 85909 >dumpcore.sh_ardusub.85909.out 2>&1
[ardusub-2] Failed
[mavros_node-8] Error:   mavconn: tcp0: receive: Connection reset by peer
[mavros_node-8]          at line 283 in ./src/tcp.cpp
[mavros_node-8] Error:   mavconn: tcp0: shutdown: Transport endpoint is not connected
[mavros_node-8]          at line 194 in ./src/tcp.cpp
[ERROR] [ardusub-2]: process has died [pid 85909, exit code -6, cmd 'ardusub -S -w -M JSON --defaults /home/orca4/colcon_ws/install/orca_bringup/share/orca_bringup/cfg/sub.parm -I0 --home 33.810313,-118.39386700000001,0.0,270.0'].
[mavros_node-8] Error:   mavconn: tcp0: send: channel closed!
[mavros_node-8]          at line 234 in ./src/tcp.cpp

What's the cause of this error? Thanks.

--- a/README.md
+++ b/README.md
@@ -41,7 +41,6 @@ See the [Dockerfile](docker/Dockerfile) for installation details.
 Install these packages:
 * [ROS2 Humble](https://docs.ros.org/en/humble/Installation.html)
 * [Gazebo Garden 7.1.0](https://gazebosim.org/docs/garden/install)
-* [ros_gz, built from source on the ros2 branch](https://github.com/gazebosim/ros_gz)
 * [ardupilot_gazebo](https://github.com/ArduPilot/ardupilot_gazebo)
 * [ArduSub](https://ardupilot.org/dev/docs/building-setup-linux.html)
 
@@ -58,6 +57,12 @@ mkdir -p ~/colcon_ws/src
 cd colcon_ws/src
 git clone https://github.com/clydemcqueen/orca4
 vcs import < orca4/workspace.repos
+~~~
+
+
+Build ros_gz for Garden, not Fortress:
+~~~
+export GZ_VERSION=garden
 ~~~
 
 Get dependencies, ignoring Gazebo Garden rosdep keys:

--- a/docker/Dockerfile
+++ b/docker/Dockerfile
@@ -110,6 +110,9 @@ COPY --chown=$USER_UID:$USER_GID workspace.repos colcon_ws/src/orca4
 WORKDIR /home/$USERNAME/colcon_ws/src
 RUN vcs import < orca4/workspace.repos
 
+# Build for Gazebo Garden
+ENV GZ_VERSION=garden
+
 # Run rosdep over workspace repos
 # Note: ros_gz asks for some Gazebo Garden keys which are not yet present in rosdistro.
 # This is fine, since Gazebo Garden has already been installed.

--- a/orca_description/models/orca4/model.sdf
+++ b/orca_description/models/orca4/model.sdf
@@ -85,12 +85,12 @@
       <mQ>0</mQ>
       <nR>0</nR>
       <!-- Second order stability derivative: -->
-      <xUU>-33.800000000000004</xUU>
-      <yVV>-54.26875</yVV>
-      <zWW>-73.37135</zWW>
-      <kPP>-4.0</kPP>
-      <mQQ>-4.0</mQQ>
-      <nRR>-4.0</nRR>
+      <xUabsU>-33.800000000000004</xUabsU>
+      <yVabsV>-54.26875</yVabsV>
+      <zWabsW>-73.37135</zWabsW>
+      <kPabsP>-4.0</kPabsP>
+      <mQabsQ>-4.0</mQabsQ>
+      <nRabsR>-4.0</nRabsR>
     </plugin>
 
     <link name="thruster1">

--- a/orca_description/models/orca4/model.sdf.in
+++ b/orca_description/models/orca4/model.sdf.in
@@ -85,12 +85,12 @@
       <mQ>0</mQ>
       <nR>0</nR>
       <!-- Second order stability derivative: -->
-      <xUU>@xUU</xUU>
-      <yVV>@yVV</yVV>
-      <zWW>@zWW</zWW>
-      <kPP>@kPP</kPP>
-      <mQQ>@mQQ</mQQ>
-      <nRR>@nRR</nRR>
+      <xUabsU>@xUabsU</xUabsU>
+      <yVabsV>@yVabsV</yVabsV>
+      <zWabsW>@zWabsW</zWabsW>
+      <kPabsP>@kPabsP</kPabsP>
+      <mQabsQ>@mQabsQ</mQabsQ>
+      <nRabsR>@nRabsR</nRabsR>
     </plugin>
 
     <link name="thruster1">

--- a/orca_description/scripts/generate_model.py
+++ b/orca_description/scripts/generate_model.py
@@ -76,12 +76,12 @@ iyy = mass / 12 * (collision_x * collision_x + collision_z * collision_z)
 izz = mass / 12 * (collision_x * collision_x + collision_y * collision_y)
 
 # 2nd order stability for the HydrodynamicsPlugin
-xUU = -0.5 * visual_y * visual_z * 0.8 * fluid_density
-yVV = -0.5 * visual_x * visual_z * 0.95 * fluid_density
-zWW = -0.5 * visual_x * visual_y * 0.95 * fluid_density
-kPP = -0.5 * 0.008 * fluid_density
-mQQ = -0.5 * 0.008 * fluid_density
-nRR = -0.5 * 0.008 * fluid_density
+xUabsU = -0.5 * visual_y * visual_z * 0.8 * fluid_density
+yVabsV = -0.5 * visual_x * visual_z * 0.95 * fluid_density
+zWabsW = -0.5 * visual_x * visual_y * 0.95 * fluid_density
+kPabsP = -0.5 * 0.008 * fluid_density
+mQabsQ = -0.5 * 0.008 * fluid_density
+nRabsR = -0.5 * 0.008 * fluid_density
 
 # Thruster placement
 thruster_x = 0.15

--- a/setup.bash
+++ b/setup.bash
@@ -22,3 +22,6 @@ export GZ_SIM_RESOURCE_PATH=$HOME/colcon_ws/src/bluerov2_ignition/models:$HOME/c
 
 # Add orca4 models and worlds
 export GZ_SIM_RESOURCE_PATH=$HOME/colcon_ws/src/orca4/orca_description/models:$HOME/colcon_ws/src/orca4/orca_description/worlds:$GZ_SIM_RESOURCE_PATH
+
+# Build ros_gz on the humble branch for Gazebo Garden
+export GZ_VERSION=garden

--- a/workspace.repos
+++ b/workspace.repos
@@ -15,8 +15,8 @@ repositories:
     url: https://github.com/ptrmu/ros2_shared
     version: master
 
-  # ros_gz supports Humble + Garden only if built from source on the ros2 branch
+  # ros_gz default on the humble branch is Fortress; export GZ_VERSION=garden to build for Garden
   ros_gz:
     type: git
     url: https://github.com/gazebosim/ros_gz
-    version: ros2
+    version: humble

